src/lib/sql/sql_translator.cpp

```diff
@@ -1117,6 +1132,50 @@ void SQLTranslator::_translate_select_groupby_having(const hsql::SelectStatement
   }
 }
 
+void SQLTranslator::_translate_set_operation(const hsql::SetOperation& set_operator) {
+  const auto& left_input_lqp = _current_lqp;
+
+  // Create a nested SQL-Translator for the right hand site of the Set-Statement to not make changes in the current one
+  SQLTranslator nested_set_translator{_use_mvcc, _external_sql_identifier_resolver_proxy, _parameter_id_allocator,
+                                      _with_descriptions, _meta_tables};
+  const auto right_input_lqp = nested_set_translator._translate_select_statement(*set_operator.nestedSelectStatement);
+
+  AssertInput(left_input_lqp->column_expressions().size() == right_input_lqp->column_expressions().size(),
+              "The size of tables connected via set operators needs to match");
+
+  // Check to see if both input lqps use the same data-type for each column
+  for (auto column_expression_idx = size_t{0}; column_expression_idx < left_input_lqp->column_expressions().size();
+       ++column_expression_idx) {
+    const auto& left_expression = left_input_lqp->column_expressions()[column_expression_idx];
+    const auto& right_expression = right_input_lqp->column_expressions()[column_expression_idx];
```

Comment:
`column_expressions()` returns by value, so you are reaching into a vector that is due to expire at the end of the line. Good thing that asan was unhappy about it, because if it worked, it would be extremely expensive. Calculating `column_expressions()` is one of the most costly LQP operations. If you calculate both column_expressions only once (including `column_expressions().size()` above), this shouldn't be a problem anymore.

This is done in other parts of the SQLTranslator too. For a bunch of reasons, it doesn't cause issues there. Nothing I would have expected you to catch just by looking at it.